plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/notation/crowsfoot/CrowsFootDiagramNotation.java

```diff
@@ -115,6 +115,29 @@ private void createTargetDecorator(PolylineConnection conn, Color bckColor, Colo
         conn.setTargetDecoration(targetDecor);
     }
 
+    private List<DBSTableIndex> getIndexSourceAttributes(
+        Collection<? extends DBSTableIndex> indexes,
+        List<ERDEntityAttribute> sourceAttr,
+        DBRProgressMonitor monitor) throws DBException {
+        List<DBSTableIndex> sourceAttributeIndexes = new ArrayList<>();
+        for (DBSTableIndex index : indexes) {
+            if (index.isPrimary()) {
+                // primary key index
+                continue;
+            }
+            if (DBUtils.isIdentifierIndex(monitor, index)) {
+                List<DBSEntityAttribute> entityIdentifierAttributes = DBUtils.getEntityAttributes(monitor, index);
+                List<DBSEntityAttribute> sourceAttributes = sourceAttr.stream().map(s -> {
+                    return s.getObject();
+                }).collect(Collectors.toList());
```

Comment:
```suggestion
                List<DBSEntityAttribute> sourceAttributes = sourceAttr.stream().map(ERDEntityAttribute::getObject).toList();
```
And maybe you want to move this variable outside the loop?

Reply:
Yeap, thansk